ui/apps/platform/src/Components/SelectSingle/SelectSingle.tsx

```diff
@@ -38,33 +41,75 @@ function SelectSingle({
 }: SelectSingleProps): ReactElement {
     const [isOpen, setIsOpen] = useState(false);
 
-    function onSelect(_event, selection) {
-        // The mouse event is not useful.
-        setIsOpen(false);
-        handleSelect(id, selection);
+    function onSelect(
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        selection: string | number | undefined
+    ) {
+        if (typeof selection === 'string') {
+            setIsOpen(false);
+            handleSelect(id, selection);
+        }
     }
 
+    function onToggle() {
+        setIsOpen(!isOpen);
+    }
+
+    // Find the display text for the selected value
+    const getDisplayText = (): string => {
+        if (!value) {
+            return placeholderText;
+        }
+
+        const selectedChild = children.find((child) => {
+            return child.props.value === value;
+        });
+
+        return (selectedChild?.props.children as string) || value;
```

Comment:
Nit: when `selectedChild?.props.children` is not already a string, we would likely end up with `[Object object]` as what is displayed.

Reply:
That's valid. We can try to add proper type checking or use a `String()` conversion here.